druid/src/main/java/org/apache/calcite/adapter/druid/DruidJsonFilter.java

```diff
@@ -321,7 +321,7 @@ private static DruidJsonFilter toBetweenDruidFilter(RexNode rexNode, RelDataType
       return null;
     }
     final boolean isNumeric = lhs.getType().getFamily() == SqlTypeFamily.NUMERIC
-        || lhs.getType().getFamily() == SqlTypeFamily.NUMERIC;
```

Comment:
@nishantmonu51 thanks can you please add a test case for this?

Reply:
done. added test @b-slim